mani_skill/envs/tasks/tabletop/poke_cube.py

```diff
@@ -0,0 +1,192 @@
+        return self.peg.pose * self.peg_head_offsets
+
+    def _initialize_episode(self, env_idx: torch.Tensor, options: dict):
+        with torch.device(self.device):
+            b = len(env_idx)
+            self.table_scene.initialize(env_idx)
+            # initialize the peg
+            peg_xyz = torch.zeros((b, 3))
+            peg_xyz = torch.rand((b, 3)) * 0.2 - 0.1
+            peg_xyz[..., 2] = self.peg_half_width
+            peg_q = [1, 0, 0, 0]
+            peg_pose = Pose.create_from_pq(p=peg_xyz, q=peg_q)
+            self.peg.set_pose(peg_pose)
+            # initialize the cube
+            cube_xyz = torch.zeros((b, 3))
+            cube_xyz = torch.rand((b, 3)) * 0.2 - 0.1
+            cube_xyz[..., 0] = peg_xyz[..., 0] + self.peg_half_length + 0.1
+            cube_xyz[..., 2] = self.cube_half_size
+            cube_q = [1, 0, 0, 0]
+            cube_pose = Pose.create_from_pq(p=cube_xyz, q=cube_q)
+            self.cube.set_pose(cube_pose)
+            # initialize the goal region
+            goal_region_xyz = cube_xyz + torch.tensor([0.05 + self.goal_radius, 0, 0])
+            goal_region_xyz[..., 2] = 1e-3
+            goal_region_q = euler2quat(0, np.pi / 2, 0)
+            goal_region_pose = Pose.create_from_pq(p=goal_region_xyz, q=goal_region_q)
+            self.goal_region.set_pose(goal_region_pose)
+
+    def _get_obs_extra(self, info: Dict):
+        obs = dict(
+            tcp_pose=self.agent.tcp.pose.raw_pose,
+        )
+
+        if self._obs_mode in ["state", "state_dict"]:
+            obs.update(
+                cube_pose=self.cube.pose.raw_pose,
+                peg_pose=self.peg.pose.raw_pose,
+                goal_pos=self.peg.pose.p,
+                tcp_to_peg_pos=self.peg.pose.p - self.agent.tcp.pose.p,
+                peg_to_cube_pos=self.cube.pose.p - self.peg.pose.p,
+                cube_to_goal_pos=self.goal_region.pose.p - self.cube.pose.p,
+                peghead_to_cube_pos=self.peg_head_pos - self.cube.pose.p,
+            )
+        return obs
+
+    def evaluate(self):
+        is_cube_placed = (
+            torch.linalg.norm(
+                self.cube.pose.p[..., :2] - self.goal_region.pose.p[..., :2], axis=1
+            )
+            < self.goal_radius
+        )
+        peg_q = self.peg_head_pose.q
+        peg_qmat = rotation_conversions.quaternion_to_matrix(peg_q)
+        peg_euler = rotation_conversions.matrix_to_euler_angles(peg_qmat, "XYZ")
+        cube_q = self.cube.pose.q
+        cube_qmat = rotation_conversions.quaternion_to_matrix(cube_q)
+        cube_euler = rotation_conversions.matrix_to_euler_angles(cube_qmat, "XYZ")
+        angle_diff = torch.abs(peg_euler[:, 2] - cube_euler[:, 2])
+        is_peg_cube_aligned = angle_diff < 0.05
+
+        head_to_cube_dist = torch.linalg.norm(
+            self.peg_head_pos[..., :2] - self.cube.pose.p[..., :2], axis=1
+        )
+        is_peg_cube_touch = head_to_cube_dist <= self.cube_half_size + 0.005
```

Comment:
please rename is_peg_cube_touch to something like is_peg_cube_close. This condition does not check correctly if they are touching. Checking closeness is fine.

Reply:
I have modified the variable names containing "touch" incorrectly, thanks for your advice.